implant/sliver/mount/mount_windows.go

```diff
@@ -0,0 +1,287 @@
+//go:build windows
+// +build windows
+
+/*
+	Sliver Implant Framework
+	Copyright (C) 2023  Bishop Fox
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package mount
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unicode/utf16"
+	"unsafe"
+
+	"github.com/bishopfox/sliver/protobuf/sliverpb"
+
+	"golang.org/x/sys/windows"
+)
+
+const (
+	pathLength             = windows.MAX_PATH + 1
+	volumePathsLength      = 32769 // 32K + 1 for a null terminator
+	universalNameInfoLevel = 1
+	remoteNameInfoLevel    = 2
+	remoteNameMaxLength    = 1024
+)
+
+type UniversalNameInfo struct {
+	UniversalName [remoteNameMaxLength]uint16
+}
+
+var driveTypeMap = map[uint32]string{
+	0: "Unknown",
+	1: "Root Path invalid (no volume mounted for path)",
+	2: "Removable",
+	3: "Fixed disk",
+	4: "Remote / network drive",
+	5: "CD-ROM",
+	6: "RAM disk",
+}
```

Comment:
We should probably do the translation in the client to avoid adding unnecessary strings in the implant that could get signatured when the obfuscator fails.

Reply:
I found a way that will leave the Protobuf as is, since that VolumeType field is used by Windows and Linux. Linux requires a string for it, so I think it makes sense leaving it alone. Changes pushed!